traits/observers/_i_observer.py

```diff
@@ -20,6 +20,21 @@ class IObserver(abc.ABC):
     user. In order to support equality and hashing on the
     ``ObserverGraph``, ``IObserver`` needs to be hashable
     and it needs to support comparison for equality.
+
+    A subclass of ``IObserver`` is often context specific, e.g. for observing
+    a trait on an HasTraits instance, for observing items in a list, etc.
+
+    Given an ``ObserverGraph``, observers are visited from the root node
+    to the leaf nodes. The first observer (root node) will be given the object
+    the user wants to observe. It knows how to obtain the ``IObservable``
+    object(s) for attaching notifiers (see ``iter_observables``). It also
+    knows what objects should be given to the downstream/children observers in
+    the same graph (see ``iter_objects``). Then the process repeats as the
+    ``ObserverGraph`` is walked. See ``add_or_remove_notifiers``.
+
+    Some observers may require additional helper observers and notifiers,
+    e.g. an observer for traits will need to respond to the ``trait_added``
+    event. See ``iter_extra_graphs``.
```

Comment:
Now this sentence may be a bit redundant given the recent change to docstring of `iter_extra_graphs`. Will remove.